### ISSUE TITLE
RegistrationHash.GetMolLayers does not distinguish atropisomers

### ISSUE THREAD

**Issue by ricrogz** [rdkit/rdkit#7367]
The registration hash should be able to distinguish between atropisomers, but that's currently not the case:

```python3
In [1]: from rdkit import Chem

In [2]: from rdkit.Chem import RegistrationHash

# This is one of the examples from the initial atropisomers PR
In [3]: m = Chem.MolFromMolFile('RP-6306_atrop1.sdf')

In [4]: b = m.GetBondWithIdx(3)

# The atropisomer was correctly identified
In [5]: b.GetStereo()
Out[5]: rdkit.Chem.rdchem.BondStereo.STEREOATROPCW

In [6]: hash1 = RegistrationHash.GetMolLayers(m, enable_tautomer_hash_v2=True)

# No atropisomer info in the hash: for SMILES, it's encoded in the CX extension, which is not part of the hash
In [7]: print(hash1)
{
<HashLayer.CANONICAL_SMILES: 1>: 'Cc1cc2c(C(N)=O)c(N)n(-c3c(C)ccc(O)c3C)c2nc1C', 
<HashLayer.ESCAPE: 2>: '', 
<HashLayer.FORMULA: 3>: 'C18H20N4O2', 
<HashLayer.NO_STEREO_SMILES: 4>: 'Cc1cc2c(C(N)=O)c(N)n(-c3c(C)ccc(O)c3C)c2nc1C', 
<HashLayer.NO_STEREO_TAUTOMER_HASH: 5>: '[C]:[C]1:[N]:[C]2:[C](:[C]:[C]:1-[CH3]):[C](:[C](:[N]):[O]):[C](:[N]):[N]:2:[C]1:[C](-[CH3]):[C]:[C]:[C](:[O]):[C]:1-[CH3]_11_0', 
<HashLayer.SGROUP_DATA: 6>: '[]', 
<HashLayer.TAUTOMER_HASH: 7>: '[C]:[C]1:[N]:[C]2:[C](:[C]:[C]:1-[CH3]):[C](:[C](:[N]):[O]):[C](:[N]):[N]:2:[C]1:[C](-[CH3]):[C]:[C]:[C](:[O]):[C]:1-[CH3]_11_0'
}

# Now, reverse the atropisomer
In [8]: b.SetStereo(Chem.BondStereo.STEREOATROPCCW)

In [9]: hash2 = RegistrationHash.GetMolLayers(m, enable_tautomer_hash_v2=True)

# No atropisomer info either
In [10]: print(hash2)
{
<HashLayer.CANONICAL_SMILES: 1>: 'Cc1cc2c(C(N)=O)c(N)n(-c3c(C)ccc(O)c3C)c2nc1C', 
<HashLayer.ESCAPE: 2>: '', 
<HashLayer.FORMULA: 3>: 'C18H20N4O2', 
<HashLayer.NO_STEREO_SMILES: 4>: 'Cc1cc2c(C(N)=O)c(N)n(-c3c(C)ccc(O)c3C)c2nc1C', 
<HashLayer.NO_STEREO_TAUTOMER_HASH: 5>: '[C]:[C]1:[N]:[C]2:[C](:[C]:[C]:1-[CH3]):[C](:[C](:[N]):[O]):[C](:[N]):[N]:2:[C]1:[C](-[CH3]):[C]:[C]:[C](:[O]):[C]:1-[CH3]_11_0', 
<HashLayer.SGROUP_DATA: 6>: '[]', 
<HashLayer.TAUTOMER_HASH: 7>: '[C]:[C]1:[N]:[C]2:[C](:[C]:[C]:1-[CH3]):[C](:[C](:[N]):[O]):[C](:[N]):[N]:2:[C]1:[C](-[CH3]):[C]:[C]:[C](:[O]):[C]:1-[CH3]_11_0'
}

# We'd want these to be different to deduplicate the atropisomers, but ...
In [11]: hash1 != hash2
Out[11]: False
```

**Comment by ricrogz** [comment]
The fix for this is trivial once https://github.com/rdkit/rdkit/pull/7301 is merged, we'll have to wait for it.
